packages/paste-core/components/date-picker/src/utils.ts

```diff
@@ -0,0 +1,18 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import {format} from 'date-fns';
+
+// Adding a time of 12pm below because date-fns only takes a Date obj, but the DatePicker
+// `value` is returned as a string that looks like: 'YYYY-MM-DD'. In order to pass the date
+// into the formatter, we need to create a new Date using the date `value`. However, when `new Date()`
+// is used, 'T00:00:00' is assigned as the time in UTC+0 then converted tothe user's locale
+// (e.g. value becomes the previous day at 7pm in EST/UTC-5), shifting the date by 1 day.
+// We're setting the time to 12pm to add padding on either side to avoid the shift from happening.
+
+// UPDATE: added '-00:00' to the end of the `time` string so
+// that Date would automatically be instantiated in UTC+0.
+
+export const returnDateFormatter = (dateValue: string, dateFormat: string): string => {
+  const time = 'T12:00:00-00:00';
```

Comment:
This would break if someone, somehow, passed in a dateValue string with time already concatenated.
Maybe consider a check like:
```js
const time = 'T12:00:00-00:00';

export const returnDateFormatter = (dateString: string, dateFormat: string): string => {
   let dateValue = dateString;
   if (!dateValue.contains("T")) {
     dateValue = new Date(dateValue.concat(time));
   }
   return format(Number(dateValue), dateFormat);
}
```

Reply:
And then add 1 test